### PR TITLE
8328562: Convert java/awt/InputMethods/DiacriticsTest/DiacriticsTest.java applet test to main

### PR DESCRIPTION
**Pull request by azvegint**
Test converted to main.
Instructions split to Windows and Linux, with some clarifications.
The test runs successfully on Windows and Linux.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8328562](https://bugs.openjdk.org/browse/JDK-8328562): Convert java/awt/InputMethods/DiacriticsTest/DiacriticsTest.java applet test to main (**Bug** - P4)

### Reviewers
 * [Tejesh R](https://openjdk.org/census#tr) (@TejeshR13 - Committer) ⚠️ Review applies to [73d2d47d](https://git.openjdk.org/jdk/pull/18388/files/73d2d47da92a5b3e82f61468d1dc37eeee94718f)
 * [Phil Race](https://openjdk.org/census#prr) (@prrace - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/18388/head:pull/18388` \
`$ git checkout pull/18388`

Update a local copy of the PR: \
`$ git checkout pull/18388` \
`$ git pull https://git.openjdk.org/jdk.git pull/18388/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 18388`

View PR using the GUI difftool: \
`$ git pr show -t 18388`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/18388.diff">https://git.openjdk.org/jdk/pull/18388.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/18388#issuecomment-2008600696)

### CODE REVIEW

File: test/jdk/java/awt/InputMethods/DiacriticsTest/DiacriticsTest.java [hunk 1]
```diff
@@ -22,29 +22,94 @@
  */
 
 
+import jdk.test.lib.Platform;
```
Comment (TejeshR13) [inline test/jdk/java/awt/InputMethods/DiacriticsTest/DiacriticsTest.java:25]: Additional line space before imports can be removed.

File: test/jdk/java/awt/InputMethods/DiacriticsTest/DiacriticsTest.java [hunk 2]
```diff
@@ -22,29 +22,94 @@
...
-  @summary Check if diacritical signs could be typed for TextArea and TextField
-  @run applet/manual=yesno DiacriticsTest.html
+ * @test
+ * @bug 8000423 7197619 8025649
+ * @summary Check if diacritical signs could be typed for TextArea and TextField
+ * @requires (os.family == "windows" | os.family == "linux")
+ * @library /java/awt/regtesthelpers /test/lib
+ * @build PassFailJFrame jdk.test.lib.Platform
+ * @run main/manual DiacriticsTest
 */
```
Comment (TejeshR13) [inline test/jdk/java/awt/InputMethods/DiacriticsTest/DiacriticsTest.java:39]: Another minor observation - One space before asterisk.

**Review by TejeshR13 — APPROVED** [review]

File: test/jdk/java/awt/InputMethods/DiacriticsTest/DiacriticsTest.java (hunk 1 above)
Comment (azvegint) [reply test/jdk/java/awt/InputMethods/DiacriticsTest/DiacriticsTest.java:24]: Removed

**Review by prrace — APPROVED** [review]
This test looks to be a good example why we should be considering a new jtreg key like
@key keyboard
or
@key locale

**Pull request closed without merging**